src/firefly/test/edu/caltech/ipac/firefly/server/query/LightCurveProcessorTest.java

```diff
@@ -188,31 +193,41 @@ protected File createPhaseFoldedTempFile() throws IOException {
      */
     static class PeriodogramAPIRequestTest extends PeriodogramAPIRequest {
 
+        private int n_peaks = 52;
+        // Fake values passed from client,
+        // see test properties irsa.gator.service.periodogram.keys
+        private final String[] reqValues = new String[]{"x=mjd", "y=w1mpro_ep", "peaks="+n_peaks, "alg=ls"};
+
+        @Override
+        public String getParam(String param) {
+            int i = 0;
+            for (String k : reqValues) {
+                if (k.startsWith(param)) {
+                    return reqValues[i].split("=")[1];
+                }
+                i++;
+            }
+            return "";
+        }
+
         @Override
         public float getPeriod() {
             return 1.345f;
         }
 
         @Override
         public String getLcSource() {
-            try {
-                URL demo = new URL("http://web.ipac.caltech.edu/staff/ejoliet/demo/AllWISE-MEP-m82-2targets-10arsecs.tbl");
-                URLConnection uc = URLDownload.makeConnection(demo);
-                URLDownload.getDataToFile(uc, rawTable);
-                return rawTable.getAbsolutePath();
-            } catch (MalformedURLException e) {
-                e.printStackTrace();
-            } catch (FailedRequestException e) {
-                e.printStackTrace();
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
-            return null;
+            return "http://web.ipac.caltech.edu/staff/ejoliet/demo/AllWISE-MEP-m82-2targets-10arsecs.tbl";
```

Comment:
I guess you misunderstood.  "http://web.ipac.caltech.edu/staff/ejoliet/demo/AllWISE-MEP-m82-2targets-10arsecs.tbl" is used as an input file to testPhaseFoldedCurve @Test.
Instead of getting the file from an external source, it should be int our firefly_test_data repository.